lib/ecto/query.ex

```diff
@@ -431,6 +431,11 @@ defmodule Ecto.Query do
     defstruct [:expr, :file, :line, params: []]
   end
 
+  defmodule ByExpr do
+    @moduledoc false
+    defstruct [:expr, :file, :line, params: [], subqueries: [], aliases: %{}]
```

Comment:
Are you sure the `aliases` field is required? I am almost sure it is about handling `selected_as`, which is not a feature we can support in group_by/order_by. It is a select only feature.

Reply:
Ah, interesting, I didn't know what it was for. I can discard the `aliases` information and remove the `aliases` key from `ByExpr` and see if there are any issues though?

Reply:
Done.